### PR TITLE
Use the official Python image as base image

### PR DESCRIPTION
**Pull request by petrhosek**
This gives more precise control over the Python version.

### CODE REVIEW

**Comment by petrhosek** [comment]
@zeroomega This is working for me locally but we should also test this in Fuchsia's infrastructure.

**Review by boomanaiden154 — COMMENTED** [review]
I'd caution against using Python 3.10 for now. See #146. This should be fixed once we go through with #128 (due to the dependency updates there), but for now we've being doing most stuff on Python 3.8/3.9.

**Comment by zeroomega** [comment]
Once this lands I can manually test it on Fuchsia infra builders.

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-    python-is-python3 \
-    python3 \
-    python3-pip \
-    && rm -rf /var/lib/apt/lists/*
+FROM python:3.9
+
 COPY . /ml-compiler-opt
-RUN python3 -m pip install -r /ml-compiler-opt/requirements.txt
+RUN pip install --no-cache-dir -r /ml-compiler-opt/requirements.txt
```
Comment (boomanaiden154) [inline Dockerfile:4]: What's the rationale here for using `--no-cache-dir`?

Comment (petrhosek) [reply Dockerfile:4]: The cache is used to store the installation files(`.whl`, etc.) of the modules that you install through pip and the source files (`.tar.gz`, etc.) neither of which are needed inside the container, so this helps to reduce the size of the layer.

Comment (boomanaiden154) [reply Dockerfile:4]: Makes perfect sense. Was thinking of the wrong file when I saw the line.

**Review by boomanaiden154 — APPROVED** [review]

**Pull request merged**